### ISSUE TITLE
Error using CLI log command to retrieve log files from S3

### ISSUE THREAD

**Issue by kgalens** [nextflow-io/nextflow#3199]
## Bug report 

Error retrieving log, stdout and stderr fields from `nextflow log` command when running on AWS Batch backend.

### Expected behavior and actual behavior

When running locally, the `nextflow log <name> -f 'log'` command returns as expected:
```
$ nextflow run hello
N E X T F L O W  ~  version 22.08.2-edge
Launching `https://github.com/nextflow-io/hello` [crazy_rubens] DSL2 - revision: 4eab81bd42 [master]
[-        ] process > sayHello -

<<< TRUNCATED >>>

$ nextflow log crazy_rubens -f 'name,log'
sayHello (3)    Hello world!

sayHello (1)    Bonjour world!

sayHello (4)    Hola world!

sayHello (2)    Ciao world!
```

If run on AWS:
nextflow.config
```
plugins {
    id 'nf-amazon'
}

// Allows the use of docker
docker.enabled = true

profiles {
  aws {
    /* AWS Configuration */
    process {
        executor = 'awsbatch'
        queue = '<<QUEUE NAME>>'
    }

    workDir = 's3://<<BUCKET NAME>>/nextflow'
    aws {
      batch {
        cliPath = "/opt/aws-cli/bin/aws"
      }
    }
  }
}
```

```
$ nextflow run hello -profile aws
N E X T F L O W  ~  version 22.08.2-edge
Launching `https://github.com/nextflow-io/hello` [nostalgic_gilbert] DSL2 - revision: 4eab81bd42 [master]
[-        ] process > sayHello -

<<< TRUNCATED >>>

$ nextflow log nostalgic_gilbert -profile aws -f 'name,log'
WARN: Unable to start plugin 'nf-amazon' required by s3://<<BUCKET NAME>>/nextflow/d2/d4d85c87863f69ef9e9abaf76675b8
Missing plugin 'nf-amazon' required to read file: s3://<<BUCKET NAME>>/nextflow/d2/d4d85c87863f69ef9e9abaf76675b8
```

### Steps to reproduce the problem

See above. This does require having an appropriately setup environment.

### Program output 

Error message received
```
$ nextflow log nostalgic_gilbert -profile aws -f 'name,log'
WARN: Unable to start plugin 'nf-amazon' required by s3://<<BUCKET NAME>>/nextflow/d2/d4d85c87863f69ef9e9abaf76675b8
Missing plugin 'nf-amazon' required to read file: s3://<<BUCKET NAME>>/nextflow/d2/d4d85c87863f69ef9e9abaf76675b8
```

### Environment 

* Nextflow version: 22.08.2-edge 
* Java version: openjdk version "17.0.4" 2022-07-19 LTS
* Operating system: Linux
* Bash version: GNU bash, version 4.2.46(2)-release (x86_64-koji-linux-gnu)

### Additional context

The commands above I ran using the `nextflow/nextflow:latest` docker container. I also replicated running with the CLI outside of docker on my mac and was able to replicate the same behavior (nextflow version 22.04.5).

**Comment by pditommaso** [comment]
Please try again the command using the option `-log .nextflow.log` as shown below and include it into this issue 

```
nextflow -log .nextflow.log log nostalgic_gilbert -profile aws -f 'name,log'
```

**Comment by kgalens** [comment]
If I run that command I receive the following error (also posted above):
```
$ nextflow -log .nextflow.log log sick_shaw -profile aws -f 'name,log'
Unknown option: -profile -- Check the available commands and options and syntax with 'help'
```
When running without the `-profile aws`, the following is the log file created:
<details>
  <summary>nextflow log file</summary>
  ```
Sep-14 13:38:03.233 [main] DEBUG nextflow.cli.Launcher - $> nextflow -c /etc/nextflow/base-nextflow.config -log .nextflow.log log sick_shaw -f name,log
Sep-14 13:38:03.511 [main] DEBUG nextflow.plugin.PluginsFacade - Using Default plugins manager
Sep-14 13:38:03.518 [main] INFO  org.pf4j.DefaultPluginStatusProvider - Enabled plugins: []
Sep-14 13:38:03.519 [main] INFO  org.pf4j.DefaultPluginStatusProvider - Disabled plugins: []
Sep-14 13:38:03.521 [main] INFO  org.pf4j.DefaultPluginManager - PF4J version 3.4.1 in 'deployment' mode
Sep-14 13:38:03.526 [main] DEBUG nextflow.cache.CacheFactory - Using Nextflow cache factory: nextflow.cache.DefaultCacheFactory
Sep-14 13:38:03.547 [main] DEBUG nextflow.util.CustomThreadPool - Creating default thread pool > poolSize: 5; maxThreads: 1000
Sep-14 13:38:03.634 [main] DEBUG nextflow.plugin.PluginsFacade - Using Default plugins manager
Sep-14 13:38:03.694 [main] DEBUG nextflow.plugin.PluginsFacade - Using Default plugins manager
Sep-14 13:38:03.695 [main] WARN  nextflow.file.FileHelper - Unable to start plugin 'nf-amazon' required by s3://<BUCKET REDACTED>/nextflow/85/e1d022320c4ac827a0aadd9988d213
java.lang.NullPointerException: Cannot invoke "nextflow.plugin.CustomPluginManager.getPlugin(String)" because "this.manager" is null
        at nextflow.plugin.PluginsFacade.isStarted(PluginsFacade.groovy:338)
        at nextflow.plugin.PluginsFacade.startIfMissing(PluginsFacade.groovy:432)
        at nextflow.plugin.Plugins.startIfMissing(Plugins.groovy:91)
        at nextflow.file.FileHelper.autoStartMissingPlugin(FileHelper.groovy:288)
        at nextflow.file.FileHelper.asPath(FileHelper.groovy:268)
        at nextflow.cli.CmdLog$TraceAdaptor.getWorkDir(CmdLog.groovy:292)
        at nextflow.cli.CmdLog$TraceAdaptor.get(CmdLog.groovy:255)
        at groovy.lang.Binding.getVariable(Binding.java:58)
        at nextflow.processor.TaskTemplateEngine$TemplateBinding.getVariable(TaskTemplateEngine.groovy:217)
        at groovy.lang.Script.getProperty(Script.java:55)
        at org.codehaus.groovy.runtime.callsite.PogoGetPropertySite.getProperty(PogoGetPropertySite.java:49)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callGroovyObjectGetProperty(AbstractCallSite.java:341)
        at Script82C80C0E1C90B7F73949B46E2E2BF18B.run(Script82C80C0E1C90B7F73949B46E2E2BF18B:1)
        at ch.grengine.BaseGrengine.run(BaseGrengine.java:947)
        at nextflow.processor.TaskTemplateEngine$ParsableTemplate$1.writeTo(TaskTemplateEngine.groovy:175)
        at nextflow.processor.TaskTemplateEngine$ParsableTemplate$1.toString(TaskTemplateEngine.groovy:190)
        at nextflow.cli.CmdLog.printRecord(CmdLog.groovy:194)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)
        at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1268)
        at groovy.lang.MetaClassImpl.invokeMethodClosure(MetaClassImpl.java:1048)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1142)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1035)
        at groovy.lang.Closure.call(Closure.java:412)
        at nextflow.cache.CacheDB.eachRecord(CacheDB.groovy:211)
        at nextflow.cli.CmdLog$_run_closure4.doCall(CmdLog.groovy:160)
        at nextflow.cli.CmdLog$_run_closure4.call(CmdLog.groovy)
        at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2359)
        at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2344)
        at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2385)
        at nextflow.cli.CmdLog.run(CmdLog.groovy:158)
        at nextflow.cli.Launcher.run(Launcher.groovy:487)
        at nextflow.cli.Launcher.main(Launcher.groovy:646)
Sep-14 13:38:03.696 [main] DEBUG nextflow.plugin.PluginsFacade - Using Default plugins manager
Sep-14 13:38:03.697 [main] ERROR nextflow.cli.Launcher - @unknown
java.lang.IllegalStateException: Missing plugin 'nf-amazon' required to read file: s3://<BUCKET REDACTED>/nextflow/85/e1d022320c4ac827a0aadd9988d213
        at nextflow.file.FileHelper.asPath(FileHelper.groovy:320)
        at nextflow.file.FileHelper.asPath(FileHelper.groovy:274)
        at nextflow.cli.CmdLog$TraceAdaptor.getWorkDir(CmdLog.groovy:292)
        at nextflow.cli.CmdLog$TraceAdaptor.get(CmdLog.groovy:255)
        at groovy.lang.Binding.getVariable(Binding.java:58)
        at nextflow.processor.TaskTemplateEngine$TemplateBinding.getVariable(TaskTemplateEngine.groovy:217)
        at groovy.lang.Script.getProperty(Script.java:55)
        at org.codehaus.groovy.runtime.callsite.PogoGetPropertySite.getProperty(PogoGetPropertySite.java:49)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callGroovyObjectGetProperty(AbstractCallSite.java:341)
        at Script82C80C0E1C90B7F73949B46E2E2BF18B.run(Script82C80C0E1C90B7F73949B46E2E2BF18B:1)
        at ch.grengine.BaseGrengine.run(BaseGrengine.java:947)
        at nextflow.processor.TaskTemplateEngine$ParsableTemplate$1.writeTo(TaskTemplateEngine.groovy:175)
        at nextflow.processor.TaskTemplateEngine$ParsableTemplate$1.toString(TaskTemplateEngine.groovy:190)
        at nextflow.cli.CmdLog.printRecord(CmdLog.groovy:194)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:107)
        at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:323)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1268)
        at groovy.lang.MetaClassImpl.invokeMethodClosure(MetaClassImpl.java:1048)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1142)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1035)
        at groovy.lang.Closure.call(Closure.java:412)
        at nextflow.cache.CacheDB.eachRecord(CacheDB.groovy:211)
        at nextflow.cli.CmdLog$_run_closure4.doCall(CmdLog.groovy:160)
        at nextflow.cli.CmdLog$_run_closure4.call(CmdLog.groovy)
        at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2359)
        at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2344)
        at org.codehaus.groovy.runtime.DefaultGroovyMethods.each(DefaultGroovyMethods.java:2385)
        at nextflow.cli.CmdLog.run(CmdLog.groovy:158)
        at nextflow.cli.Launcher.run(Launcher.groovy:487)
        at nextflow.cli.Launcher.main(Launcher.groovy:646)
```
</details>

**Comment by pditommaso** [comment]
Solved by f5a5245b
